internal/factsengine/gatherers/hostsfile_test.go

```diff
@@ -0,0 +1,120 @@
+package gatherers
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type HostsFileTestSuite struct {
+	suite.Suite
+	fileSystem afero.Fs
+}
+
+func TestHostsFileTestSuite(t *testing.T) {
+	suite.Run(t, new(HostsFileTestSuite))
+}
+
+func (suite *HostsFileTestSuite) SetupTest() {
+	suite.fileSystem = afero.NewMemMapFs()
+
+	err := suite.fileSystem.MkdirAll("/etc", 0644)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func (suite *HostsFileTestSuite) TestHostsFileBasic() {
+	testFile, _ := os.Open("../../../test/fixtures/gatherers/hosts.basic")
+	confFile, _ := ioutil.ReadAll(testFile)
+	err := afero.WriteFile(suite.fileSystem, "/etc/hosts", confFile, 0644)
+	assert.NoError(suite.T(), err)
+	c := NewHostsFileGatherer(suite.fileSystem)
+
+	factRequests := []FactRequest{
+		{
+			Name:     "hosts_localhost",
+			Gatherer: "hosts",
+			Argument: "localhost",
+			CheckID:  "check1",
+		},
+		{
+			Name:     "hosts_somehost",
+			Gatherer: "hosts",
```

Comment:
Once again, how does this fit in the DSL representation of the involved check (1.5.1)?

Reply:
facts:
  - name: nodes_hosts_file_resolution
    gatherer: hosts_file
    argument: "node01:node02"

expectations:
  - name: nodes_resolvable_using_hosts_file
	# not sure how to express that the resulting list should have at least 2 results, just a suggestion
    expect: len(nodes_hosts_file_resolution) > 1

Reply:
With the new expression evaluation in place, we can now use e.g. `array.some(|x| x == expected_value)` to check for the presence of an IP in the obtained facts, so it should be fine. If we don't know such IP we can just check for the length of the resulting array.